### PR TITLE
Refactor serializer

### PR DESCRIPTION
**Pull request by Yash-Vardhan-Sharma**
## Description
wrote and update serializers

## Relevant Issue
Issue #Tag the issue here

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/RecoEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/RecoEdge/blob/main/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/RecoEdge/labels)
- [ ] My changes are covered by tests

### CODE REVIEW

File: fedrec/serialization/abstract_serializer.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-
 from abc import ABC, abstractmethod
 from xml.dom import NotSupportedErr
```
Comment (vkkhare) [inline fedrec/serialization/abstract_serializer.py:2]: update this line as well

File: fedrec/serialization/job_response_serializer.py [hunk 2]
```diff
@@ -12,24 +12,18 @@ def __init__(self, serialization_strategy):
...
-        worker_state = self.deserialize_attribute(obj.workerstate)
+        senderid = self.deserialize_attribute(obj['senderid'])
+        receiverid = self.deserialize_attribute(obj['senderid'])
+        
 
         return JobResponseMessage(obj["job_type"],
-                                  job_args,
-                                  job_kwargs,
-                                  worker_state)
+                                  senderid,
```
Comment (vkkhare) [inline fedrec/serialization/job_response_serializer.py:28]: it needs to serialise result as well

File: fedrec/serialization/trainer_state_serializer.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    def __init__(self, serialization_strategy):
+        super().__init__(serialization_strategy)
+
+    def serialize(self, obj):
+ 
+        response_dict = {}
+        response_dict["id"] = obj.id
+        response_dict["round_idx"] = obj.round_idx
+        response_dict["state_dict"] = self.serialize_attribute(
```
Comment (ramesht007) [inline fedrec/serialization/trainer_state_serializer.py:17]: Write serializer for state dict

File: fedrec/serialization/trainer_state_serializer.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            obj.local_sample_number)
+        response_dict["local_training_steps"] = self.serialize_attribute(
+            obj.local_training_steps)
+
+        return self.serialization_strategy.unparse(response_dict)
+
+    def deserialize(self, obj: Dict):
+
+        state_dict = self.deserialize_attribute(
+            obj['state_dict'])
```
Comment (ramesht007) [inline fedrec/serialization/trainer_state_serializer.py:32]: same as above, write deserializer

File: fedrec/serialization/trainer_state_serializer.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        model_preproc = self.deserialize_attribute(
```
Comment (ramesht007) [inline fedrec/serialization/trainer_state_serializer.py:33]: same as above, write deserializer

File: fedrec/serialization/trainer_state_serializer.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            obj['model_preproc'])
+        local_sample_number = self.deserialize_attribute(
```
Comment (ramesht007) [inline fedrec/serialization/trainer_state_serializer.py:35]: don't need to deserialize, we can send it directly

File: fedrec/serialization/aggregator_state_serializer.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        response_dict["storage"] = obj.storage
+        response_dict["in_neighbours"] = self.serialize_attribute(
+            obj.in_neighbours)
+        response_dict["out_neighbours"] = self.serialize_attribute(
+            obj.out_neighbours)
+
+        return self.serialization_strategy.unparse(response_dict)
+
+    def deserialize(self, obj: Dict):
+
```
Comment (vkkhare) [inline fedrec/serialization/aggregator_state_serializer.py:28]: call json.parse first before deserializing to get the object dictionary

File: fedrec/serialization/job_response_serializer.py [hunk 8]
```diff
@@ -14,24 +14,20 @@ def __init__(self, serialization_strategy):
     def serialize(self, obj):
         response_dict = {}
         response_dict["job_type"] = obj.job_type
-        response_dict["job_args"] = [self.serialize_attribute(arg)
-                                     for arg in obj.obj_args]
-        response_dict["job_kwargs"] = [self.serialize_attribute(kwarg)
-                                       for kwarg in obj.obj_kwargs]
-        response_dict["worker_state"] = self.serialize_attribute(
-            obj.workerstate)
+        response_dict["senderid"] = self.serialize_attribute(
```
Comment (vkkhare) [inline fedrec/serialization/job_response_serializer.py:17]: These don't need to be serialized/ deserialized. The ids are string or int

File: fedrec/serialization/job_response_serializer.py [hunk 9]
```diff
@@ -14,24 +14,20 @@ def __init__(self, serialization_strategy):
...
         return self.serialization_strategy.unparse(response_dict)
 
     def deserialize(self, obj: Dict):
-        job_type = obj.job_type
-        job_args = [self.deserialize_attribute(arg) for
-                    arg in obj.obj_args]
-        job_kwargs = [self.deserialize_attribute(kwarg) for
-                      kwarg in obj.obj_kwargs]
-        worker_state = self.deserialize_attribute(obj.workerstate)
+        senderid = self.deserialize_attribute(obj['senderid'])
```
Comment (vkkhare) [inline fedrec/serialization/job_response_serializer.py:27]: parse from serialization strategy first

File: fedrec/serialization/trainer_state_serializer.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        response_dict["storage"] = obj.storage
+        response_dict["model_preproc"] = self.serialize_attribute(
+            obj.model_preproc)
+        response_dict["local_sample_number"] = obj.local_sample_number
+        response_dict["local_training_steps"] = obj.local_training_steps
+
+        return self.serialization_strategy.unparse(response_dict)
+
+    def deserialize(self, obj: Dict):
+
```
Comment (vkkhare) [inline fedrec/serialization/trainer_state_serializer.py:28]: parse from serializaiton strategy first

**Review by vkkhare — CHANGES_REQUESTED** [review]

File: fedrec/serialization/aggregator_state_serializer.py (hunk 7 above)
Comment (Yash-Vardhan-Sharma) [reply fedrec/serialization/aggregator_state_serializer.py:29]: done

File: fedrec/serialization/job_response_serializer.py (hunk 9 above)
Comment (Yash-Vardhan-Sharma) [reply fedrec/serialization/job_response_serializer.py]: done

File: fedrec/serialization/trainer_state_serializer.py (hunk 10 above)
Comment (Yash-Vardhan-Sharma) [reply fedrec/serialization/trainer_state_serializer.py:29]: done
